### PR TITLE
[ParameterUpdate] Fix `Parameterized` synthesis for types nested in functions.

### PR DESCRIPTION
**Pull request by dan-zheng**
Addresses [SR-8360](https://bugs.swift.org/browse/SR-8360).

### CODE REVIEW

**Comment by dan-zheng** [comment]
@swift-ci Please test tensorflow

File: lib/Sema/DerivedConformanceParameterized.cpp [hunk 1]
```diff
@@ -306,7 +306,6 @@ deriveParameterized_allParameters(DerivedConformance &derived) {
   auto *setterDecl =
       derived.declareDerivedPropertySetter(derived.TC, allParamsDecl, returnTy);
   setterDecl->setBodySynthesizer(&derivedBody_allParametersSetter);
-  allParamsDecl->setSetterAccess(nominal->getFormalAccess());
```
Comment (rxwei) [inline lib/Sema/DerivedConformanceParameterized.cpp]: What if it's within a public struct? The synthesized struct should be public there.

Comment (dan-zheng) [reply lib/Sema/DerivedConformanceParameterized.cpp]: You bring up a good point about access level handling.

However, the deleted code set the access level of the `allParameters` computed property setter, which is not related to the synthesized `Parameters` struct.

(There's code somewhere that makes the accessors of `allParameters` have the same access level as the `allParameters` declaration itself. The deleted code caused problems, I added it early on during experimentation.)

When I run `swiftc -dump-ast` on the following code:
```swift
import TensorFlow
public struct Model : Parameterized {
  @TFParameter var w: Float
}
```
I can see that both `Parameters` and `allParameters` are public, following the `Model` struct. (Note that `w` is actually internal, which is expected.)

```
(struct_decl range=[foo.swift:2:8 - line:4:1] "Model" interface type='Model.Type' access=public non-resilient inherits: Parameterized
  (var_decl range=[foo.swift:3:20 - line:3:20] "w" type='Float' interface type='Float' access=internal ...
  (struct_decl implicit "Parameters" interface type='Model.Parameters.Type' access=public non-resilient inherits: ParameterAggregate
  (var_decl implicit "allParameters" type='Model.Parameters' interface type='Model.Parameters' access=public ...
```

Comment (dan-zheng) [reply lib/Sema/DerivedConformanceParameterized.cpp]: Perhaps we can verify access levels by adding `-dump-ast` to the `RUN` line and adding `CHECK` comments. I'd prefer to do that in a separate PR though.

Comment (rxwei) [reply lib/Sema/DerivedConformanceParameterized.cpp]: `w` being internal should imply that `Parameter.w` is internal as well.

Comment (dan-zheng) [reply lib/Sema/DerivedConformanceParameterized.cpp]: Good point. This is the case ([relevant code](https://github.com/dan-zheng/swift/blob/dbf87db73b682084037cba087ada06371dfe9dd0/lib/Sema/DerivedConformanceParameterized.cpp#L361)):
```
(struct_decl implicit "Parameters" interface type='Model.Parameters.Type' access=public non-resilient inherits: ParameterAggregate
  (var_decl "w" type='Float' interface type='Float' access=internal readImpl=stored writeImpl=stored readWriteImpl=stored)
```

**Review by rxwei — APPROVED** [review]

**Pull request merged**